### PR TITLE
arbiter source code

### PR DESCRIPTION
**Pull request by kathleendaily**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,94 @@
-Server Name
+Server
 ============
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/f056b46535d441d18d9d7a78916f4f86)](https://www.codacy.com?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=propelmarketing/server&amp;utm_campaign=Badge_Grade) [![CircleCI](https://circleci.com/gh/propelmarketing/server.svg?style=svg&circle-token=16667f90ebd396fea6203aa9876f73592bfa668c)](https://circleci.com/gh/propelmarketing/server)
 
 ## Description
 
+Login with facebook:
```
Comment (gooftroop) [inline README.md]: We can get rid of the description content (keep the title)

File: appspec.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#   instructional comments and can be safely left in the file or
+#   ignored.
+# For help completing this file, see the "AppSpec File Reference" in the
+#   "AWS CodeDeploy User Guide" at
+#   http://docs.aws.amazon.com/codedeploy/latest/userguide/app-spec-ref.html
+version: 0.0
+os: linux
+files:
+  - source: /
+    destination: /home/ec2-user/server
```
Comment (gooftroop) [inline appspec.yml]: make this file appspec.yml.template

File: circle.yml [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+    - yarn test:coverage
+    - cat ./coverage/lcov.info | ./node_modules/.bin/codacy-coverage
+    - source ./scripts/selenium.sh
+deployment:
+  production:
+    tag: /v[0-9]+(\.[0-9]+)*/
+    commands:
+      - yarn build
+      - source ./scripts/ec2/deploy_prod.sh
+  qa:
```
Comment (gooftroop) [inline circle.yml]: lets remove the qa section

File: config/custom-environment-variables.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  "passport": {
```
Comment (gooftroop) [inline config/custom-environment-variables.json]: remove passport

File: config/custom-environment-variables.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "hostname": "SERVER_HOSTNAME",
+    "port": "SERVER_PORT",
+    "secure": "SERVER_SECURE",
+    "assets": {
+      "cdn": "TH_CDN",
+      "staticRoot": "TH_CDN_STATIC_ROOT",
+      "templateEngine": "SERVER_VIEW_ENGINE",
+      "views": "SERVER_VIEWS"
+    },
+    "session": {
```
Comment (gooftroop) [inline config/custom-environment-variables.json]: remove session

File: config/default.json [hunk 6]
```diff
@@ -1,35 +1,120 @@
 {
-  "loggers": [
-    {
-      "name": "root",
-      "level": "debug"
+  "auth": {
```
Comment (gooftroop) [inline config/default.json]: remove auth

File: config/default.json [hunk 7]
```diff
@@ -1,35 +1,120 @@
...
-    "http": {
-      "backlog": 511,
-      "hostname": "localhost",
-      "port": 3000
+      "cdn": "//s3.amazonaws.com/th-static-cdn",
+      "staticRoot": "static",
+      "templateEngine": "ejs",
+      "views": "src/views"
     },
+    "allowedClientNames": [
```
Comment (gooftroop) [inline config/default.json]: remove

File: config/default.json [hunk 8]
```diff
@@ -1,35 +1,120 @@
...
+      "thrivehive",
+      "portal"
+    ],
+    "agencyBlacklist": [
+      "localhost",
+      "127.0.0.1"
+    ],
     "session": {
```
Comment (gooftroop) [inline config/default.json]: remove

Comment (gooftroop) [inline config/default.json]: for everything being remove in default.json, remove in the remaining config files

File: config/sequelize.json [hunk 9]
```diff
@@ -1,29 +1,52 @@
 {
```
Comment (gooftroop) [inline config/sequelize.json]: delete sequelize.json

File: config/sequelize.json.template [hunk 10]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (gooftroop) [inline config/sequelize.json.template]: ditto

File: migrations/20170601162705-create-users-table.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
+'use strict';
```
Comment (gooftroop) [inline migrations/20170601162705-create-users-table.js]: delete the migrations directroy

File: newrelic.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+ */
+exports.config = {
+  /**
+   * Array of application names.
+   */
+  app_name: ['Server'],
+  /**
+   * Your New Relic license key.
+   */
+  license_key: '027c7518b17857c7ce4c6216c65f21275d7dcf45',
```
Comment (gooftroop) [inline newrelic.js]: remove the key value and just leave empty string

File: newrelic.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+ * New Relic agent configuration.
+ *
+ * See lib/config.default.js in the agent distribution for a more complete
+ * description of configuration variables and their potential values.
+ */
+exports.config = {
+  /**
+   * Array of application names.
+   */
+  app_name: ['Server'],
```
Comment (gooftroop) [inline newrelic.js]: leave the array empty - "server" is just a place holder until the implementing app replaces it with the app name

File: package.json [hunk 14]
```diff
@@ -1,50 +1,86 @@
 {
-  "name": "<name>",
+  "name": "server",
   "version": "1.0.0",
-  "description": "ThriveHive Microservice",
+  "description": "ThriveHive CAS",
```
Comment (gooftroop) [inline package.json]: replace with Mircoserivce

File: package.json [hunk 15]
```diff
@@ -1,50 +1,86 @@
 {
-  "name": "<name>",
+  "name": "server",
```
Comment (gooftroop) [inline package.json]: put server in angle brackets

File: package.json [hunk 16]
```diff
@@ -1,50 +1,86 @@
...
+    "test:coverage": "yarn cross-env NODE_ENV=test yarn nyc yarn mocha-webpack",
+    "test:windows": "yarn cross-env NODE_ENV=test yarn mocha-webpack",
+    "watch": "concurrently --kill-other --raw \"yarn build-watch\" \"pm2 startOrRestart process.dev.json\"",
+    "watch:log": "concurrently --kill-other --raw \"yarn build-watch\" \"pm2 logs server --lines 0 startOrRestart process.dev.json\"",
+    "build:watch": "yarn build && yarn watch"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/propelmarketing/<name>.git"
+    "url": "git+https://github.com/propelmarketing/server.git"
```
Comment (gooftroop) [inline package.json]: ditto on the brackets

File: package.json [hunk 17]
```diff
@@ -1,50 +1,86 @@
...
   },
   "keywords": [
     "nodejs",
+    "CAS",
+    "oauth2",
```
Comment (gooftroop) [inline package.json]: remove both

File: package.json [hunk 18]
```diff
@@ -1,50 +1,86 @@
...
     "express"
   ],
   "author": "dev@thrivehive.com",
-  "license": "2017 ThriveHive, All rights reserved",
+  "license": "UNLICENSED",
   "bugs": {
-    "url": "https://github.com/propelmarketing/<name>/issues"
+    "url": "https://github.com/propelmarketing/server/issues"
```
Comment (gooftroop) [inline package.json]: brackets

File: package.json [hunk 19]
```diff
@@ -1,50 +1,86 @@
...
   },
-  "homepage": "https://github.com/propelmarketing/<name>#readme",
+  "homepage": "https://github.com/propelmarketing/server#readme",
```
Comment (gooftroop) [inline package.json]: brackets

File: package.json [hunk 20]
```diff
@@ -1,50 +1,86 @@
...
   "dependencies": {
+    "@newrelic/native-metrics": "^2.1.1",
+    "babel-polyfill": "^6.23.0",
+    "bcrypt": "^1.0.2",
```
Comment (gooftroop) [inline package.json]: remove

File: package.json [hunk 21]
```diff
@@ -1,50 +1,86 @@
...
     "bunyan": "^1.8.10",
-    "concurrently": "^3.1.0",
+    "bunyan-cloudwatch": "^2.2.0",
+    "bunyan-newrelic-stream": "^1.2.0",
+    "codacy-coverage": "^2.0.2",
+    "concurrently": "^3.4.0",
     "config": "^1.24.0",
-    "cookie-parser": "^1.4.3",
+    "connect-flash-plus": "^0.2.1",
+    "connect-redis": "^3.3.0",
```
Comment (gooftroop) [inline package.json]: remove

File: package.json [hunk 22]
```diff
@@ -1,50 +1,86 @@
...
+    "ejs": "^2.5.6",
     "express": "^4.14.0",
     "express-session": "^1.15.3",
+    "generate-password": "^1.3.0",
```
Comment (gooftroop) [inline package.json]: remove

File: package.json [hunk 23]
```diff
@@ -1,50 +1,86 @@
...
     "lumber-cli": "^1.3.1",
+    "newrelic": "^1.39.1",
+    "passport": "^0.3.2",
```
Comment (gooftroop) [inline package.json]: remove all passport

File: package.json [hunk 24]
```diff
@@ -1,50 +1,86 @@
...
+    "passport-google-oauth": "^1.0.0",
+    "passport-http": "^0.3.0",
+    "passport-http-bearer": "^1.0.1",
+    "passport-instagram": "^1.0.0",
+    "passport-local": "^1.0.0",
+    "passport-oauth2-client-password": "^0.1.2",
+    "passport-twitter": "^1.0.4",
     "pg": "^6.1.5",
     "pg-hstore": "^2.3.2",
+    "request-promise-native": "^1.0.4",
```
Comment (gooftroop) [inline package.json]: remove

File: package.json [hunk 25]
```diff
@@ -1,50 +1,86 @@
...
+    "safe-json-stringify": "^1.0.4",
+    "sendgrid": "^5.1.1",
     "sequelize": "4.0.0-2",
     "sequelize-cli": "^2.7.0",
-    "underscore": "^1.8.3",
-    "validator": "^7.0.0"
+    "swagger-express-middleware": "^1.0.0-alpha.12",
+    "uuid": "^3.1.0",
+    "validator": "^7.0.0",
+    "zxcvbn": "^4.4.2"
```
Comment (gooftroop) [inline package.json]: remove

File: package.json [hunk 26]
```diff
@@ -67,17 +104,20 @@
...
     "mocha": "^3.2.0",
     "mocha-webpack": "^0.7.0",
+    "nyc": "^11.0.2",
+    "pm2": "^2.5.0",
     "sinon": "^2.3.2",
     "sinon-chai": "^2.8.0",
     "source-map-support": "^0.4.15",
     "style-loader": "^0.13.1",
     "supertest": "^3.0.0",
+    "swagger": "^0.7.5",
```
Comment (gooftroop) [inline package.json]: swagger should be in prod deps

File: process.dev.json [hunk 27]
```diff
@@ -0,0 +1,13 @@
+{
+  "apps": [{
+    "script": "index.js",
+    "name": "server",
```
Comment (gooftroop) [inline process.dev.json]: brackets

File: process.json [hunk 28]
```diff
@@ -0,0 +1,11 @@
+{
+  "apps": [{
+    "script": "index.js",
+    "name": "server",
```
Comment (gooftroop) [inline process.json]: brackets

File: scripts/ec2/after_install.sh [hunk 29]
```diff
@@ -0,0 +1,46 @@
...
+# Load the production config
+cd ~/server/config
+CONFIG_FILE="local.json"
+S3_URI="s3://th-config/server/$DEPLOYMENT_GROUP_NAME"
+aws s3 cp "$S3_URI/$CONFIG_FILE" "$CONFIG_FILE"
+if [ ! -f "$CONFIG_FILE" ]; then
+  echo "Failed to download the correct config ($CONFIG_FILE) file!" && exit 1
+fi
+
+# Load the prod-level sequelize config if in prod
```
Comment (gooftroop) [inline scripts/ec2/after_install.sh]: remove the commented code

File: scripts/ec2/deploy_qa.sh [hunk 30]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
```
Comment (gooftroop) [inline scripts/ec2/deploy_qa.sh]: remove file

File: scripts/ec2/start.sh [hunk 31]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# EC2 ApplicationStart hook bash script
+
+source ~/.bash_profile
+
+if [ ! -z "$DEPLOYMENT_GROUP_NAME" ]; then
+ export NODE_ENV=$DEPLOYMENT_GROUP_NAME
+fi
+
+cd ~/server
```
Comment (gooftroop) [inline scripts/ec2/start.sh]: brackets. Just to make it easier - replace all instances of server that you replaced arbiter with to be <server>

File: src/server.js [hunk 32]
```diff
@@ -3,27 +3,37 @@
...
+
+import swaggerApi from 'configuration/swagger.yml';
+
+// Enable Newrelic Reporting if the server is in a production environment
+let newrelic: Object | null = null;
+if (process.env.NODE_ENV === 'production') {
+  newrelic = require('newrelic');
+}
 
-import ErrorMiddleware from 'server/middleware/error';
```
Comment (gooftroop) [inline src/server.js]: keep error, index, logger and static middleware both here and the source files. Also keep the source files in exception (but clear out errors specific to passport or specific input/parameters - leave the generalized stuff), the logger util, etc. anything else that generalized and not specific to just arbiter

File: package.json (hunk 15 above)
Comment (kathleendaily) [inline package.json]: brackets are invalid in 'name' and will throw an error when running the server

File: config/sequelize.json (hunk 9 above)
Comment (kathleendaily) [inline config/sequelize.json]: github is still showing this and all the other files i deleted for some reason

File: static/scripts/passwordResetValidation.js [hunk 33]
```diff
@@ -0,0 +1,94 @@
+(function () {
```
Comment (gooftroop) [inline static/scripts/passwordResetValidation.js]: we can remove this file

File: static/scripts/loginValidation.js [hunk 34]
```diff
@@ -0,0 +1,104 @@
+var form = (function() {
```
Comment (gooftroop) [inline static/scripts/loginValidation.js]: we can remove this file

File: static/scripts/login.js [hunk 35]
```diff
@@ -0,0 +1,101 @@
+var api = (function () {
```
Comment (gooftroop) [inline static/scripts/login.js]: we can remove this file

File: static/scripts/forgotPasswordValidation.js [hunk 36]
```diff
@@ -0,0 +1,33 @@
+$("#forgotPasswordForm").submit(function(event) {
```
Comment (gooftroop) [inline static/scripts/forgotPasswordValidation.js]: we can remove this file

File: src/views/pages/error/500.ejs [hunk 37]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <link rel="stylesheet" src="<%= cdn %>/<%= agency %>/css/styles.css"></link>
```
Comment (gooftroop) [inline src/views/pages/error/500.ejs]: my bad - it's actually `style.css`. Might want to update the other files too

File: src/server/middleware/index.js [hunk 38]
```diff
@@ -38,11 +43,80 @@ export default class AbstractMiddleware {
  * @param  {Function} next     [description]
  * @return {[type]}            [description]
  */
-const ensureLoggedIn = function(request: Object, response: Object, next: Function): void {
```
Comment (gooftroop) [inline src/server/middleware/index.js]: What happened to ensureLoggedIn? We actually want to keep the function (leave it empty) and keep validateApiAccess, but we can get rid of validateUserSession

File: src/server.js [hunk 39]
```diff
@@ -110,36 +182,71 @@ export default class Server {
...
-   *
+   * [stop description]
+   * @param  {Function} callback [description]
+   * @return {[type]}            [description]
    */
   stop(callback: Function | null = null): void {
-    if (this.app && this.app.server) {
-      this.app.server.close(() => {
+    if (this.app && this.app.<server>) {
+      this.app.<server>.close(() => {
```
Comment (gooftroop) [inline src/server.js]: in this case server is actually an attribute on app provided by express, so any reference to `this.app.server` can be left as-is

File: src/server.js [hunk 40]
```diff
@@ -110,36 +182,71 @@ export default class Server {
...
     }
 
-    const httpConfig = this.config.get('http');
-    this.app.listen(httpConfig.get('port'), httpConfig.get('hostname'), httpConfig.get('backlog'), () => {
+    const cb = () => {
       if (callback != null) {
         callback();
       }
-      const message = `Server listening at ${httpConfig.get('hostname')}:${httpConfig.get('port')}...`;
+      const message = `<server> listening at ${this.config.get('hostname')}:${this.config.get('port')}...`;
```
Comment (gooftroop) [inline src/server.js]: Original was correct - we only want `<server>` when it was originally `arbiter`

File: src/server.js [hunk 41]
```diff
@@ -110,36 +182,71 @@ export default class Server {
...
+        return next();
+      }
+      return response.redirect(`https://${request.hostname}:${this.config.get('port')}${request.url}`);
+    });
+    const sslConfig = this.config.get('ssl');
+    const httpsConfig = Object.assign({}, sslConfig, {
+      key: fs.readFileSync(sslConfig.get('key')),
+      cert: fs.readFileSync(sslConfig.get('cert'))
     });
+    return https.create<server>(httpsConfig, this.app)
```
Comment (gooftroop) [inline src/server.js]: ummm what's going on here?

File: src/server.js [hunk 42]
```diff
@@ -110,36 +182,71 @@ export default class Server {
    */
   start(callback: Function | null = null): void {
     if (!this.app) {
-      throw new Error('Cannot start server: the express instance is not defined');
+      throw new Error('Cannot start <server>: the express instance is not defined');
```
Comment (gooftroop) [inline src/server.js]: original was correct

File: src/server.js [hunk 43]
```diff
@@ -3,46 +3,70 @@
...
 
   /**
-   *
+   * [constructor description]
+   * @param  {[type]} void [description]
+   * @return {[type]}      [description]
    */
   constructor(): void {
-    this.config = config.get('server');
+    this.config = config.get('<server>');
```
Comment (gooftroop) [inline src/server.js]: original was correct

File: src/server.js [hunk 44]
```diff
@@ -3,46 +3,70 @@
...
+import process from 'process';
+import swaggerMiddleware from 'swagger-express-middleware';
 
-import ErrorMiddleware from 'server/middleware/error';
-import SessionMiddleware from 'server/middleware/session';
-import StaticMiddleware from 'server/middleware/static';
+import swaggerApi from 'configuration/swagger.yml';
 
-import IndexController from 'server/controllers';
+import AgencyMiddleware from '<server>/middleware/agency';
```
Comment (gooftroop) [inline src/server.js]: all imports should be `server` rather than `<server>` since the directory is named `server`

File: src/server.js [hunk 45]
```diff
@@ -3,46 +3,70 @@
...
+// Allow config mutations
+process.env.ALLOW_CONFIG_MUTATIONS = true;
 
 /**
- *
+ * [app description]
+ * @type {[type]}
  */
-export default class Server {
+export default class <server> {
```
Comment (gooftroop) [inline src/server.js]: Original was correct

File: scripts/ec2/start.sh [hunk 46]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# EC2 ApplicationStart hook bash script
+
+source ~/.bash_profile
+
+if [ ! -z "$DEPLOYMENT_GROUP_NAME" ]; then
+ export NODE_ENV=$DEPLOYMENT_GROUP_NAME
+fi
+
+cd ~/<server>
```
Comment (gooftroop) [inline scripts/ec2/start.sh]: 🌮  this is what `<server>` is correct for

File: package.json [hunk 47]
```diff
@@ -1,50 +1,80 @@
...
+    "passport-instagram": "^1.0.0",
+    "passport-local": "^1.0.0",
+    "passport-oauth2-client-password": "^0.1.2",
+    "passport-twitter": "^1.0.4",
     "pg": "^6.1.5",
     "pg-hstore": "^2.3.2",
+    "safe-json-stringify": "^1.0.4",
+    "sendgrid": "^5.1.1",
     "sequelize": "4.0.0-2",
     "sequelize-cli": "^2.7.0",
```
Comment (gooftroop) [inline package.json]: we can remove sequelize imports

File: package.json [hunk 48]
```diff
@@ -1,50 +1,80 @@
...
+    "concurrently": "^3.4.0",
     "config": "^1.24.0",
-    "cookie-parser": "^1.4.3",
+    "connect-flash-plus": "^0.2.1",
+    "ejs": "^2.5.6",
     "express": "^4.14.0",
     "express-session": "^1.15.3",
     "lumber-cli": "^1.3.1",
+    "newrelic": "^1.39.1",
+    "passport": "^0.3.2",
```
Comment (gooftroop) [inline package.json]: remove all passport libs

File: package.json [hunk 49]
```diff
@@ -1,50 +1,80 @@
...
+    "passport-facebook": "^2.1.1",
+    "passport-google-oauth": "^1.0.0",
+    "passport-http": "^0.3.0",
+    "passport-http-bearer": "^1.0.1",
+    "passport-instagram": "^1.0.0",
+    "passport-local": "^1.0.0",
+    "passport-oauth2-client-password": "^0.1.2",
+    "passport-twitter": "^1.0.4",
     "pg": "^6.1.5",
```
Comment (gooftroop) [inline package.json]: remove pg libs

File: package.json [hunk 50]
```diff
@@ -1,50 +1,80 @@
...
     "pg-hstore": "^2.3.2",
+    "safe-json-stringify": "^1.0.4",
+    "sendgrid": "^5.1.1",
```
Comment (gooftroop) [inline package.json]: remove sendgrid lib

File: package.json [hunk 51]
```diff
@@ -1,50 +1,80 @@
...
+    "bunyan-newrelic-stream": "^1.2.0",
+    "codacy-coverage": "^2.0.2",
+    "concurrently": "^3.4.0",
     "config": "^1.24.0",
-    "cookie-parser": "^1.4.3",
+    "connect-flash-plus": "^0.2.1",
+    "ejs": "^2.5.6",
     "express": "^4.14.0",
     "express-session": "^1.15.3",
     "lumber-cli": "^1.3.1",
```
Comment (gooftroop) [inline package.json]: remove lumber

File: package.json [hunk 52]
```diff
@@ -1,50 +1,80 @@
...
+    "bunyan-cloudwatch": "^2.2.0",
+    "bunyan-newrelic-stream": "^1.2.0",
+    "codacy-coverage": "^2.0.2",
+    "concurrently": "^3.4.0",
     "config": "^1.24.0",
-    "cookie-parser": "^1.4.3",
+    "connect-flash-plus": "^0.2.1",
+    "ejs": "^2.5.6",
     "express": "^4.14.0",
     "express-session": "^1.15.3",
```
Comment (gooftroop) [inline package.json]: remove session

File: config/webpack/webpack.js [hunk 53]
```diff
@@ -54,26 +55,29 @@ module.exports = {
...
+      __CLIENT__: false,
+      __SERVER__: true,
+      __PRODUCTION__: false,
+      __DEV__: true
     })
   ],
   output: {
     chunkFilename: '[name].[id].js',
     filename: 'index.js',
-    library: 'Server',
```
Comment (gooftroop) [inline config/webpack/webpack.js]: The original actually is fine

File: config/webpack/webpack.js [hunk 54]
```diff
@@ -4,7 +4,7 @@ const nodeExternals = require('webpack-node-externals');
 
 // This is not particularly robust...
 const root = process.cwd();
-const entry = 'src/server.js';
+const entry = 'src/<server>.js';
```
Comment (gooftroop) [inline config/webpack/webpack.js]: original was correct

File: config/production.json [hunk 55]
```diff
@@ -1,23 +1,42 @@
...
-    "http": {
-      "hostname": "",
-      "port": 80
-    },
-    "session": {
-      "secret": ""
-    }
+    "hostname": "0.0.0.0"
+  },
+  "sendgrid": {
```
Comment (gooftroop) [inline config/production.json]: remove sendgrid

File: README.md [hunk 56]
```diff
@@ -1,28 +1,47 @@
...
+  ```
+- Must be able to clone from the <server> github repo
+- A ssh key has been generated specifically for deployment access, the keys has been uploaded to CircleCI, and the public key has been added to the target server's ~/.ssh/authorized_keys
+- From a local machine, run the following commands to initialize the pm2 environment on the target server:
+  ```
+  ./node_modules/.bin/pm2 deploy <environment_key> setup
+  ```
+  Where <environment_key> is either production or qa, depending on the purpose of the server
+
 ### Admin Console
```
Comment (gooftroop) [inline README.md]: Remove admin section

File: README.md [hunk 57]
```diff
@@ -1,28 +1,47 @@
-Server Name
+<server>
 ============
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/f056b46535d441d18d9d7a78916f4f86)](https://www.codacy.com?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=propelmarketing/server&amp;utm_campaign=Badge_Grade) [![CircleCI](https://circleci.com/gh/propelmarketing/<server>.svg?style=svg&circle-token=16667f90ebd396fea6203aa9876f73592bfa668c)](https://circleci.com/gh/propelmarketing/<server>)
```
Comment (gooftroop) [inline README.md]: Remove badges

File: test/server/controllers/root.spec.js [hunk 58]
```diff
@@ -0,0 +1,19 @@
...
+import config from 'config';
+import express from 'express';
+import request from 'supertest';
+import sinon from 'sinon';
+
+import * as RootController from 'server/controllers/root';
+
+describe('RootController', function () {
+  it('tests that redirectIndex calls \'response.redirect\' to \'/login\'', function () {
+
```
Comment (kathleendaily) [inline test/server/controllers/root.spec.js]: added it back in

File: config/docker/docker-compose.dev.yml [hunk 59]
```diff
@@ -0,0 +1,18 @@
...
+      dockerfile: config/docker/Dockerfile.dev
+    # depends_on:
+    #   - db
+  db:
+    image: postgres
+    container_name: $DB_CONTAINER_NAME
+    network_mode: "service:<server>"
+    environment:
+      - DB_CONTAINER_NAME=${DB_CONTAINER_NAME}
+      - POSTGRES_USER=<server>
```
Comment (gooftroop) [inline config/docker/docker-compose.dev.yml]: let's normalize this to `postgres` for all future use

File: config/docker/docker-compose.dev.yml [hunk 60]
```diff
@@ -0,0 +1,18 @@
...
+      - POSTGRES_PASSWORD=ThriveHive10
+      - POSTGRES_DB=<server>
```
Comment (gooftroop) [inline config/docker/docker-compose.dev.yml]: Let's make this `$APPLICATION`

File: config/docker/docker-compose.yml [hunk 61]
```diff
@@ -0,0 +1,29 @@
...
+  <server>:
+    image: "<server>-$NODE_ENV"
+    container_name: $CONTAINER_NAME
+    privileged: true
+    network_mode: "bridge"
+    build:
+      context: ../../
+      dockerfile: config/docker/Dockerfile.prod
+      args:
+        - APPLICATION=<server>
```
Comment (gooftroop) [inline config/docker/docker-compose.yml]: I think we can remove `=<server>`

File: scripts/devops/docker/__main__.py [hunk 62]
```diff
@@ -0,0 +1,110 @@
...
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "action",
+    choices=[ "start", "init" ]
+)
+
+parser.add_argument(
+    "application",
+    choices=[ "<server>", "ubuntu" ]
```
Comment (gooftroop) [inline scripts/devops/docker/__main__.py]: my bad! let's add `arbiter` back

**Pull request merged**